.github/workflows/feature-branch-cleanup.yaml

```diff
@@ -0,0 +1,69 @@
+name: Feature Branch Cleanup
+
+on:
+  workflow_dispatch:
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  cleanup:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check branch name
+        run: |
+          if [[ "${GITHUB_REF_NAME}" == "main" ]] || ! [[ "${GITHUB_REF_NAME}" =~ ^[0-9]+-.* ]]; then
+            echo "Feature branch deployment can not be initiated on branch '$GITHUB_REF_NAME'" >> $GITHUB_STEP_SUMMARY
```

Comment:
Did you mean `Feature branch cleanup`? 😇

Reply:
Oops, nice catch 😄